app/lib/features/chat_ng/widgets/events/text_message_event.dart

```diff
@@ -8,23 +11,47 @@ import 'package:flutter_matrix_html/text_parser.dart';
 class TextMessageEvent extends StatelessWidget {
   final String roomId;
   final MsgContent content;
-  final bool isUser;
-  final bool nextMessageGroup;
-  final bool wasEdited;
+  final Map<String, dynamic> metadata;
 
   const TextMessageEvent({
     super.key,
     required this.content,
-    required this.isUser,
-    required this.nextMessageGroup,
     required this.roomId,
-    this.wasEdited = false,
+    required this.metadata,
   });
 
   @override
   Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+    final chatTheme = Theme.of(context).chatTheme;
+    final colorScheme = Theme.of(context).colorScheme;
     final body = content.formattedBody() ?? content.body();
 
+    String? msgType = metadata['msgType'];
+    bool isUser = metadata['isUser'];
+    bool isNotice = (msgType == 'm.notice' || msgType == 'm.server_notice');
+    bool nextMessageGroup = metadata['nextMessageGroup'];
+    bool wasEdited = metadata['wasEdited'];
```

Comment:
as stated above. let's just make that proper typed params to put in. there are several here that if removed from the sender would lead to _runtime_ errors...